v0/destinations/hs/HSTransform-v2.js

```diff
@@ -0,0 +1,486 @@
+          metadata.push(ev.metadata);
+        }
+      });
+
+      // CRM_CREATE_CUSTOM_OBJECTS has objectType value
+      // extract objectType from the end of the endpoint
+      const objectType = chunk[0].message.endpoint.split("/").pop();
+
+      batchEventResponse.batchedRequest.endpoint = BATCH_CREATE_CUSTOM_OBJECTS.replace(
+        ":objectType",
+        objectType
+      );
+    }
+
+    batchEventResponse.batchedRequest.body.JSON = {
+      inputs: identifyResponseList
+    };
+
+    if (batchOperation === "createContacts") {
+      batchEventResponse.batchedRequest.endpoint = BATCH_IDENTIFY_CRM_CREATE_NEW_CONTACT;
+    } else if (batchOperation === "updateContacts") {
+      batchEventResponse.batchedRequest.endpoint = BATCH_IDENTIFY_CRM_UPDATE_NEW_CONTACT;
+    }
+
+    batchEventResponse.batchedRequest.headers = message.headers;
+    batchEventResponse.batchedRequest.params = message.params;
+
+    batchEventResponse = {
+      ...batchEventResponse,
+      metadata,
+      destination
+    };
+    batchedResponseList.push(
+      getSuccessRespEvents(
+        batchEventResponse.batchedRequest,
+        batchEventResponse.metadata,
+        batchEventResponse.destination,
+        true
+      )
+    );
+  });
+
+  return batchedResponseList;
+};
+
+const batchEvents = destEvents => {
+  let batchedResponseList = [];
+  const trackResponseList = [];
+  // create contact chunck
+  const createContactEventsChunk = [];
+  // update contact chunk
+  const updateContactEventsChunk = [];
+  // general indentify event chunk
+  const eventsChunk = [];
+  destEvents.forEach(event => {
+    // handler for track call
+    // track call does not have batch endpoint
+    if (event.message.messageType === "track") {
+      const { message, metadata, destination } = event;
+      const endpoint = get(message, "endpoint");
+
+      const batchedResponse = defaultBatchRequestConfig();
+      batchedResponse.batchedRequest.headers = message.headers;
+      batchedResponse.batchedRequest.endpoint = endpoint;
+      batchedResponse.batchedRequest.body = message.body;
+      batchedResponse.batchedRequest.params = message.params;
+      batchedResponse.batchedRequest.method =
+        defaultGetRequestConfig.requestMethod;
+      batchedResponse.metadata = [metadata];
+      batchedResponse.destination = destination;
+
+      trackResponseList.push(
+        getSuccessRespEvents(
+          batchedResponse.batchedRequest,
+          batchedResponse.metadata,
+          batchedResponse.destination
+        )
+      );
+    } else if (event.message.operation === "createContacts") {
+      // Identify: making chunks for CRM create contact endpoint
+      createContactEventsChunk.push(event);
+    } else if (event.message.operation === "updateContacts") {
+      // Identify: making chunks for CRM update contact endpoint
+      updateContactEventsChunk.push(event);
+    } else {
+      eventsChunk.push(event);
```

Comment:
Here the chunks should be made based on type of objects

Reply:
we are not as its rETL and in rETL only one object type will be present at a time